source/adapters/level_zero/context.cpp

```diff
@@ -574,6 +583,9 @@ ur_event_handle_t ur_context_handle_t_::getEventFromContextCache(
   if (Event->CounterBasedEventsEnabled != CounterBasedEventEnabled) {
     return nullptr;
   }
+  if (Event->InterruptBasedEventsEnabled != InterruptBasedEventEnabled) {
```

Comment:
This will not work unless the interrupt based events are somehow first on the Cache container. The same is true of the counter-based events. There needs to be separate pool for interrupt-driven events (or maybe they shouldn't be cached at all?).

Reply:
good catch, added to getEventCache to address interrupt case.

Reply:
So, just to confirm, now there is a separate cache for both counting events and interrupt based events correct?

If the cache retrieved must contain only the types of events that can be used ie counter based events etc, then the if checks are not needed. ie a counter based event cache would retrieve counter based events, interrupt based event cache will have interrupt based events.

Reply:
> So, just to confirm, now there is a separate cache for both counting events and interrupt based events correct?
> 
> If the cache retrieved must contain only the types of events that can be used ie counter based events etc, then the if checks are not needed. ie a counter based event cache would retrieve counter based events, interrupt based event cache will have interrupt based events.

So after the rebase, getEventCache now returns the cached event for the given profiling mode and provided scope. I simply added another case of interrupt based. As for the eventPoolCache, I've already added another pool for interrupt based. 
Like you said, we need to make sure counter based event cache retrieves counter based events and likewise for interrupt based event. That's why this check is in place (Event->InterruptBasedEventsEnabled != InterruptBasedEventEnabled).

Reply:
Alternatively, we can always remove the caching of interrupt based events. If this is what we want, I can simply remove this portion of the patch.